### ISSUE TITLE
Can I download one file from multiple mirror link ?

### ISSUE THREAD

**Issue by jannson** [cavaliercoder/grab#36]
Here is a new requirement, and the case is:
a large file on the internet which has multiple mirror download links, and the md5sum is the same.
the mirror http server all support accept-range.
than I want to use grab this file from multiple mirror link to speed up download bandwidth.

**Comment by cavaliercoder** [comment]
This is a great feature request, though difficult to implement. I'll look into it. Submission welcome.

**Comment by jannson** [comment]
I can try a demo for this case , but I cannot support it directly in Grab, for the Grab implement is quit complex.

**Comment by cavaliercoder** [comment]
Yes it is complicated I apologize :( I work very hard to make it simply but the are many possible states and transitions. This project was a surprise to me!

For anyone considering a pull request, please see notes in #46 first.
